repository/managers.py

```diff
@@ -55,115 +103,129 @@ def group_items(self, groupId, page=1):
-                'title': item_data.get('item', {}).get('title'),
-                'authors': item_data.get('item', {}).get('authors', []),
-                'itemType': item_data.get('item', {}).get('itemType'),
-                'addedOn': item_data.get('item', {}).get('dateAdded', 'Unknown date'),
-                'url': item_data.get('item', {}).get('url')
-            }
-
-            # Extract Giles upload details if available
-            giles_uploads = item_data.get('item', {}).get('gilesUploads', [])
-
-            if giles_uploads:
-                giles_details = []
-                extracted_text = giles_uploads[0].get('extractedText', {})
-
-                if extracted_text and extracted_text.get('content-type') == 'text/plain':
-                    extracted_text_data = get_giles_document_details(self.user, extracted_text.get('id'))
-                    item_data['item']['text'] = extracted_text_data
-                elif giles_uploads[0].get('pages'):
-                    pages = giles_uploads[0].get('pages')
-                    text = ""
-                    for page in pages:
-                        if page.get('text') and page.get('text').get('content-type') == 'text/plain':
-                            data = get_giles_document_details(self.user, page.get('text').get('id'))
-                            text += data
-                    item_data['item']['text'] = text
-                else:
-                    item_data['item']['text'] = "No valid text/plain content found."
-            else:
-                print("No Giles uploads available")
-                item_data['item']['text'] = "No Giles uploads available."
-
-            item_data['item']['details'] = item_details
-
-            return item_data
-
-        else:
-            response.raise_for_status()
+        if not giles_uploads:
+            return "No Giles uploads available."
+
+        try:
+            upload = giles_uploads[0]
+            text_content = ""
+
+            # Extract plain text if available
+            extracted_text = upload.get('extractedText', {})
+            if extracted_text and extracted_text.get('content-type') == 'text/plain':
+                text_content = get_giles_document_details(self.user, extracted_text['id'])
+                if text_content is None:
+                    raise CitesphereAPIError(message="Failed to fetch document text from Giles, please try again later.", error_code="GILES_TEXT_ERROR", details="Failed to fetch document text from Giles")
+
+            # Fallback to extracting text from pages
+            elif 'pages' in upload:
+                for page in upload['pages']:
+                    text_data = page.get('text')
+                    if text_data and text_data.get('content-type') == 'text/plain':
+                        page_text = get_giles_document_details(self.user, text_data['id'])
+                        if page_text is not None:
+                            text_content += page_text
```

Comment:
I think I made a comment somehwere, that this doesn't make much sense?

Reply:
Yes, this is in [bug/VOGRE-27](https://github.com/diging/vogon-resuscitate/pull/18). I have to modularize the giles endpoint and also show files that are still being processed.